File: java/of/of9-StackQueue.java
// 2022-04-20 02:26:57
// Runtime 1150ms(6.9%)
// Memory 48mb(69.0%)

class CQueue {
    Stack<Integer> in;
    Stack<Integer> out;
    Integer ret;
    public CQueue() {
        this.in = new Stack();
        this.out = new Stack();
    }
    
    public void appendTail(int value) {
        this.in.push(value);
    }
    
    public int deleteHead() {
        if(this.in.empty())
            return -1;
        while(!this.in.empty())
            this.out.push(this.in.pop());
        this.ret = this.out.pop();
        while(!this.out.empty())
            this.in.push(this.out.pop());
        return ret.intValue();
    }
}

/**
 * Your CQueue object will be instantiated and called as such:
 * CQueue obj = new CQueue();
 * obj.appendTail(value);
 * int param_2 = obj.deleteHead();
 */

File: py3/of9-StackQueue.py
# 2022-04-19 23:33:45
# Runtime 1316ms(10.9%)
# Memory 19mb(57.8%)

##
# 10.9%应该是有人用list作弊吧，
# 应该
# 毕竟这个题除了把入栈反过来好像也没有其他解法了
###

class CQueue:

    def __init__(self):
        self.storage = list()
        self.reverse_out = list()

    def appendTail(self, value: int) -> None:
        self.storage.append(value)

    def deleteHead(self) -> int:
        if not self.storage:
            return -1
        while self.storage:
            self.reverse_out.append(self.storage.pop(-1))
        ret = self.reverse_out.pop(-1)
        while self.reverse_out:
            self.storage.append(self.reverse_out.pop(-1))
        return ret


# Your CQueue object will be instantiated and called as such:
# obj = CQueue()
# obj.appendTail(value)
# param_2 = obj.deleteHead()
